presto-hive/src/main/java/com/facebook/presto/hive/util/InternalHiveSplitFactory.java

```diff
@@ -252,4 +256,32 @@ private static boolean pathMatchesPredicate(Optional<Domain> pathDomain, String
 
         return pathDomain.get().includesNullableValue(utf8Slice(path));
     }
+
+    private static boolean infoColumnsMatchPredicates(Map<Integer, Domain> constraints,
+                                                      String path,
+                                                      long fileSize,
+                                                      long fileModificationTime)
+    {
+        if (constraints.isEmpty()) {
+            return true;
```

Comment:
Should this be false instead of true ? I'm not super familiar with this code. But when would constraints be empty ? Would be great to have a test for this situation.

Reply:
Constraints would be empty when there are no filters on info columns in the query. We should return true, so the split will be scheduled to worker. Any simple SELECT query without filters or filters on non-info columns will already be testing this scenario.